src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -532,6 +532,18 @@ public void startJenkins() throws Throwable {
                 port = readPort(portFile);
             }
             if (port != 0) {
+
+                // Currently this file is created when a runRemotely is executed (CustomJenkinsRule constructor)
+                // But this file is needed before finishes startJenkins in order to be accessible by external systems
```

Comment:
```suggestion
                // But this file is needed before startJenkins finishes in order to be accessible by external systems
```

But the test asserts that it is present _after_ `startJenkins` finishes. So I do not understand what it is you are trying to fix. Or what these “external systems” would be—this XML file is a private implementation detail of Jenkins core, intended to be read and written only by that process.